### PR TITLE
Firing guns scares and/or angers all nearby mobs.

### PR DESCRIPTION
**Pull request by Phylogeny**
By setting the revenge target of all nearby living entities to the player firing a gun, hostile mobs like spiders and pigmen will attack the shooter, and peaceful mobs will panic and flee from the shooter. This also adds a function to your newly added silencers.

The silenced/unsilenced aggro ranges can be changes to whatever you want. And of course, feel free to reject this PR outright if you don't like the idea.

### CODE REVIEW

**Comment by Phylogeny** [comment]
Also, I could easily add a config to enable/disable this mechanic, as well as the silenced/unsilenced ranges, if you want.

**Comment by Phylogeny** [comment]
I just went ahead and added the configs, as I can always revert the commit if you don't want it configurable.

**Comment by Phylogeny** [comment]
When villagers are aggroed, they spawn particles. Since it seems odd to me to do so when fireing a gun around them, I added a entity class exemption list config pre-loaded with villagers. If you like the other configs, but want this reverted, let me know.

File: src/main/java/com/mrcrayfish/guns/item/ItemGun.java [hunk 1]
```diff
@@ -136,6 +147,27 @@ public static void fire(World worldIn, EntityPlayer playerIn, ItemStack heldItem
...
+				dx = x - entity.posX;
+				dy = y - entity.posY;
+				dz = z - entity.posZ;
+				if (dx * dx + dy * dy + dz * dz <= r)
+				{
+					entity.setRevengeTarget(playerIn);
+				}
+			}
+		}
+
```
Comment (Phylogeny) [inline src/main/java/com/mrcrayfish/guns/item/ItemGun.java]: The above code is an optimized version of the following code:
```double r = silenced ? 10 : 20;
Vec3d pos = playerIn.getPositionVector().addVector(0.5, 0.5, 0.5);
AxisAlignedBB box = new AxisAlignedBB(pos, pos).grow(r);
r *= r;
for (EntityLivingBase entity : playerIn.world.getEntitiesWithinAABB(EntityLivingBase.class, box))
{
	if (entity.getDistanceSq(pos.x, pos.y, pos.z) <= r)
	{
		entity.setRevengeTarget(playerIn);
	}
}

**Comment by Phylogeny** [comment]
Since setting a mob as its own revenge target only has the effect of inducing panic in peaceful mobs, I added a config for disabling the default behavior of hostile mobs targeting shooters.

**Comment by MrCrayfish** [comment]
You have outdone yourself dude! I am definitely going to merge this. Are you able to resolve the conflicts?

**Comment by Phylogeny** [comment]
Glad you like it. Yeah, I'll resolve them now.

**Comment by MrCrayfish** [comment]
RIP conflicts

**Pull request merged**